### ISSUE TITLE
0.6.0: test suite is failing

### ISSUE THREAD

**Issue by kloczek** [maoschanz/drawing#268]
```
+ /usr/bin/meson test -C noarch-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/drawing-0.6.0/noarch-redhat-linux-gnu'
ninja: no work to do.
1/3 Validate desktop file   OK             0.01s
2/3 Validate appstream file FAIL           1.79s (exit status 1)
3/3 Validate schema file    OK             0.01s

Ok:                 2
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0


The output from the failed tests:

2/3 Validate appstream file FAIL           1.79s (exit status 1)

--- command ---
13:27:05 /usr/bin/appstream-util validate data/com.github.maoschanz.drawing.appdata.xml
--- stdout ---
data/com.github.maoschanz.drawing.appdata.xml: FAILED:
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_menu.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_menu.png]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_new.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_new.png]
• style-invalid         : <caption> is too long [The "new image" menu opened while the selection is being scaled up];longest allowed is 50 chars
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_selection.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_selection.png]
• style-invalid         : <image> has vertical padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_tools_preview.png]
• style-invalid         : <image> has horizontal padding [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_tools_preview.png]
• style-invalid         : <caption> is too long [The thumbnail preview, showing various shapes zoomed in];longest allowed is 50 chars
--- stderr ---
Validation of files failed
-------
```

**Comment by maoschanz** [comment]
the test suite didn't block the flatpak packaging and release so i'm not sure what's going on

my theory is that the flatpak runtime doesn't have `appstream-util` so this test

```meson
appstream_util = find_program('appstream-util', required: false)
if appstream_util.found()
	test(
		'Validate appstream file',
		appstream_util,
		args: ['validate', appstream_file]
	)
endif
```
just isn't runned at all

----

**or** it is runned using `appstreamcli` which does work:

```
~/Projets/drawing$ appstreamcli validate data/com.github.maoschanz.drawing.appdata.xml.in 
I: com.github.maoschanz.drawing:6: nonstandard-gnome-extension kudos
I: com.github.maoschanz.drawing:17: description-first-para-too-short "Drawing" is a basic image editor, supporting PNG, JPEG and BMP file types.

Validation was successful: infos: 2
~/Projets/drawing$
```

**or** the flatpak runtime has some configuration such that "validate" means "validate-relax"?

```
~/Projets/drawing$ appstream-util validate-relax data/com.github.maoschanz.drawing.appdata.xml.in 
data/com.github.maoschanz.drawing.appdata.xml.in: OK
~/Projets/drawing$
```````

**Comment by maoschanz** [comment]
i'd suggest you uninstall appstream-util and see if something improves

**Comment by kloczek** [comment]
If that means that test suite communicates with some external systems to obrtain some data over appstream-util I see some general issue: many build systems are completly cut off from public network and mene as well. I'm still testing new version I did not build new drawing in such env but ..

**Comment by maoschanz** [comment]
>many build systems are completly cut off from public network

Yes it's a problem that the PPA maintainer reported to me, and indeed if i try to validate with appstream-util in plane mode, it complains about things like

```
~/Projets/drawing$ appstream-util validate data/com.github.maoschanz.drawing.appdata.xml.in 
data/com.github.maoschanz.drawing.appdata.xml.in: FAILED:
• url-not-found         : <screenshot> failed to connect: Cannot resolve hostname [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_menu.png]
• url-not-found         : <screenshot> failed to connect: Cannot resolve hostname [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_new.png]
• style-invalid         : <caption> is too long [The "new image" menu opened while the selection is being scaled up];longest allowed is 50 chars
• url-not-found         : <screenshot> failed to connect: Cannot resolve hostname [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_selection.png]
• url-not-found         : <screenshot> failed to connect: Cannot resolve hostname [https://raw.githubusercontent.com/maoschanz/drawing/master/docs/screenshots/0.6/gnome_tools_preview.png]
• style-invalid         : <caption> is too long [The thumbnail preview, showing various shapes zoomed in];longest allowed is 50 chars
Validation of files failed
~/Projets/drawing$
```

but i haven't change that part at all... The test was the same on 0.4 https://github.com/maoschanz/drawing/blob/0.4/data/meson.build#L23

These images had less paddings and shorter captions on versions 0.4.x but at least it looks like most build systems had an internet connection until now 🤔

Or people didn't read CONTRIBUTING.md so they didn't add appstream-util to the build deps 😅

**Comment by kloczek** [comment]
To be honest I really do not undestand idea behiind appstream metadata and I'm closer and closer to completly remove those files from my distro.
Theoretically those files should describe something to instal. Problem is that those matadata are availabnle only when for example drawing actually **IS** installed and usually installation process is performed over package manager.
Those files are creating yet anothet version of somethhing which acttually is is provided over package manager.

**Comment by maoschanz** [comment]
>Those files are creating yet anothet version of somethhing which acttually is is provided over package manager.

yes but it's cross-distro, so the developer is writing it himself, which is better than expecting all package maintainers to provide all the accurate data + write a full description of the app + the release notes + taking up-to-date screenshots

>Theoretically those files should describe something to instal. Problem is that those matadata are availabnle only when for example drawing actually IS installed

are they? maybe it depends on how you install things, but 95% of these informations are used in GNOME Software, and the release notes are the only info i don't know how to access before installing

----

If it can help, with the debian packaging system it's possible to override the step where it runs the tests https://github.com/maoschanz/drawing/blob/master/debian/rules#L7 and it works, maybe you can do it too?

**Comment by maoschanz** [comment]
Ok i've fixed it, you can build the tag 0.6.0b. It's the exact same as 0.6.0 except that screenshots have a hardcoded black background

**Comment by kloczek** [comment]
Yep 0.6.0b is OK now.
Thx :)
